### ISSUE TITLE
v3.0: "missing dependency" and unexpected behaviour

### ISSUE THREAD

**Issue by 75lb** [npm/npm#8742]
hi.. i'm on node 0.12.5 and npm 3.0.0.. if i install this package:
```sh
$ npm i column-layout
```
my tree looks like this (notice the unmet dep)
```
$ npm ls
npm-3-bug@1.0.0 /Users/lloyd/Documents/75lb/gists/npm-3-bug
└─┬ column-layout@1.0.0
  ├─┬ array-tools@2.0.0-5
  │ ├─┬ ansi-escape-sequences@2.1.0
  │ │ └── array-tools@1.8.6
  │ └─┬ object-tools@2.0.1
  │   └── UNMET DEPENDENCY array-tools@^1.8.4
  ├─┬ command-line-args@0.5.9
  │ ├─┬ array-tools@1.8.6
  │ │ └── typical@2.2.0
  │ ├─┬ handlebars@3.0.3
  │ │ ├─┬ optimist@0.6.1
  │ │ │ ├── minimist@0.0.10
  │ │ │ └── wordwrap@0.0.3
  │ │ ├─┬ source-map@0.1.43
  │ │ │ └── amdefine@0.1.1
  │ │ └─┬ uglify-js@2.3.6
  │ │   ├── async@0.2.10
  │ │   └── optimist@0.3.7
  │ ├─┬ handlebars-ansi@0.2.0
  │ │ └─┬ ansi-escape-sequences@1.0.2
  │ │   └── array-tools@1.8.6
  │ ├─┬ nature@0.5.7
  │ │ ├─┬ array-tools@1.8.6
  │ │ │ └── typical@2.2.0
  │ │ └── typical@1.0.0
  │ └── typical@1.0.0
  ├─┬ object-tools@1.6.7
  │ └── array-tools@1.8.6
  ├── string-tools@0.1.8
  ├── transform-tools@0.1.1
  ├── typical@2.2.0
  └── word-wrap@1.0.3

npm ERR! missing: array-tools@^1.8.4, required by object-tools@2.0.1
```

if i try to `npm install object-tools` i get this oddness (notice the package i'm installing has appeared as `object-tools@1.6.7 extraneous` underneath the unmet dep, three layers deep).. 
```
$ npm ls --depth 10
npm-3-bug@1.0.0 /Users/lloyd/Documents/75lb/gists/npm-3-bug
└─┬ column-layout@1.0.0
  ├─┬ array-tools@2.0.0-5
  │ ├─┬ ansi-escape-sequences@2.1.0
  │ │ └── array-tools@1.8.6
  │ └─┬ object-tools@2.0.1
  │   ├─┬ array-tools@1.8.6
  │   │ └── UNMET DEPENDENCY object-tools@^1.6.1
  │   └── object-tools@1.6.7 extraneous
  ├─┬ command-line-args@0.5.9
  │ ├─┬ array-tools@1.8.6
  │ │ └── typical@2.2.0
  │ ├─┬ handlebars@3.0.3
  │ │ ├─┬ optimist@0.6.1
  │ │ │ ├── minimist@0.0.10
  │ │ │ └── wordwrap@0.0.3
  │ │ ├─┬ source-map@0.1.43
  │ │ │ └── amdefine@0.1.1
  │ │ └─┬ uglify-js@2.3.6
  │ │   ├── async@0.2.10
  │ │   └── optimist@0.3.7
  │ ├─┬ handlebars-ansi@0.2.0
  │ │ └─┬ ansi-escape-sequences@1.0.2
  │ │   └── array-tools@1.8.6
  │ ├─┬ nature@0.5.7
  │ │ ├─┬ array-tools@1.8.6
  │ │ │ └── typical@2.2.0
  │ │ └── typical@1.0.0
  │ └── typical@1.0.0
  ├─┬ object-tools@1.6.7
  │ └── array-tools@1.8.6
  ├── string-tools@0.1.8
  ├── transform-tools@0.1.1
  ├── typical@2.2.0
  └── word-wrap@1.0.3

npm ERR! extraneous: object-tools@1.6.7 /Users/lloyd/Documents/75lb/gists/npm-3-bug/node_modules/array-tools/node_modules/object-tools/node_modules/object-tools
npm ERR! missing: object-tools@^1.6.1, required by array-tools@1.8.6
```

**Comment by 75lb** [comment]
as an FYI, this is how the tree looks since i rolled back to npm 2.12.0
```
$ npm ls
command-line-usage@0.0.0 /Users/lloyd/Documents/75lb/command-line-usage
└─┬ column-layout@1.0.0
  ├─┬ array-tools@2.0.0-5
  │ ├─┬ ansi-escape-sequences@2.1.0
  │ │ └─┬ array-tools@1.8.6
  │ │   └── object-tools@1.6.7
  │ └─┬ object-tools@2.0.1
  │   └─┬ array-tools@1.8.6
  │     └── object-tools@1.6.7
  ├─┬ command-line-args@0.5.9
  │ ├─┬ array-tools@1.8.6
  │ │ └── typical@2.2.0
  │ ├─┬ handlebars@3.0.3
  │ │ ├─┬ optimist@0.6.1
  │ │ │ ├── minimist@0.0.10
  │ │ │ └── wordwrap@0.0.3
  │ │ ├─┬ source-map@0.1.43
  │ │ │ └── amdefine@0.1.1
  │ │ └─┬ uglify-js@2.3.6
  │ │   ├── async@0.2.10
  │ │   └─┬ optimist@0.3.7
  │ │     └── wordwrap@0.0.3
  │ ├─┬ handlebars-ansi@0.2.0
  │ │ └── ansi-escape-sequences@1.0.2
  │ ├── nature@0.5.7
  │ └── typical@1.0.0
  ├─┬ object-tools@1.6.7
  │ └── array-tools@1.8.6
  ├── string-tools@0.1.8
  ├── transform-tools@0.1.1
  ├── typical@2.2.0
  └── word-wrap@1.0.3
```

**Comment by iarna** [comment]
Thiis, seems to be fixed in npm@3.2.0. Can you give it a try for yourself?

```
rebecca@Caldina:~/code/npmtest/8742$ npm install --save column-layout@1.0.0
8742@1.0.0 /Users/rebecca/code/npmtest/8742
└─┬ column-layout@1.0.0
  ├─┬ array-tools@2.0.1
  │ ├─┬ ansi-escape-sequences@2.1.1
  │ │ └── collect-all@0.1.0
  │ ├── array-back@1.0.2
  │ ├─┬ collect-json@1.0.0
  │ │ ├── stream-connect@0.1.0
  │ │ └── stream-via@0.1.0
  │ ├─┬ object-tools@2.0.3
  │ │ ├── collect-json@0.1.1
  │ │ └── test-value@0.1.0
  │ ├── sort-array@0.1.0
  │ └── test-value@1.0.1
  ├─┬ command-line-args@0.5.9
  │ ├─┬ array-tools@1.8.6
  │ │ └── typical@2.3.0
  │ ├─┬ handlebars@3.0.3
  │ │ ├─┬ optimist@0.6.1
  │ │ │ ├── minimist@0.0.10
  │ │ │ └── wordwrap@0.0.3
  │ │ ├─┬ source-map@0.1.43
  │ │ │ └── amdefine@1.0.0
  │ │ └─┬ uglify-js@2.3.6
  │ │   ├── async@0.2.10
  │ │   └── optimist@0.3.7
  │ ├─┬ handlebars-ansi@0.2.0
  │ │ └─┬ ansi-escape-sequences@1.0.2
  │ │   └── array-tools@1.8.6
  │ ├─┬ nature@0.5.7
  │ │ ├─┬ array-tools@1.8.6
  │ │ │ └── typical@2.3.0
  │ │ └── typical@1.0.0
  │ └── typical@1.0.0
  ├─┬ object-tools@1.6.7
  │ └── array-tools@1.8.6
  ├── string-tools@0.1.8
  ├── transform-tools@0.1.1
  ├── typical@2.3.0
  └── word-wrap@1.1.0
rebecca@Caldina:~/code/npmtest/8742$ npm ls
8742@1.0.0 /Users/rebecca/code/npmtest/8742
└─┬ column-layout@1.0.0
  ├─┬ array-tools@2.0.1
  │ ├─┬ ansi-escape-sequences@2.1.1
  │ │ └── collect-all@0.1.0
  │ ├── array-back@1.0.2
  │ ├─┬ collect-json@1.0.0
  │ │ ├── stream-connect@0.1.0
  │ │ └── stream-via@0.1.0
  │ ├─┬ object-tools@2.0.3
  │ │ ├── collect-json@0.1.1
  │ │ └── test-value@0.1.0
  │ ├── sort-array@0.1.0
  │ └── test-value@1.0.1
  ├─┬ command-line-args@0.5.9
  │ ├─┬ array-tools@1.8.6
  │ │ └── typical@2.3.0
  │ ├─┬ handlebars@3.0.3
  │ │ ├─┬ optimist@0.6.1
  │ │ │ ├── minimist@0.0.10
  │ │ │ └── wordwrap@0.0.3
  │ │ ├─┬ source-map@0.1.43
  │ │ │ └── amdefine@1.0.0
  │ │ └─┬ uglify-js@2.3.6
  │ │   ├── async@0.2.10
  │ │   └── optimist@0.3.7
  │ ├─┬ handlebars-ansi@0.2.0
  │ │ └─┬ ansi-escape-sequences@1.0.2
  │ │   └── array-tools@1.8.6
  │ ├─┬ nature@0.5.7
  │ │ ├─┬ array-tools@1.8.6
  │ │ │ └── typical@2.3.0
  │ │ └── typical@1.0.0
  │ └── typical@1.0.0
  ├─┬ object-tools@1.6.7
  │ └── array-tools@1.8.6
  ├── string-tools@0.1.8
  ├── transform-tools@0.1.1
  ├── typical@2.3.0
  └── word-wrap@1.1.0
```

**Comment by 75lb** [comment]
hiya, yes it is working now.. although i'm not 100% sure whether it's due to a code fix or the dep tree of column-layout changing... but thanks, this particular case is no longer an issue :+1:
